app/controllers/posts_controller.rb

```diff
@@ -1,9 +1,29 @@
 class PostsController < ApplicationController
   def index
-    @posts = Post.all
+    @user = User.find(params[:user_id])
+    @posts = @user.posts
   end
 
   def show
+    @user = User.find(params[:user_id])
     @post = Post.find(params[:id])
   end
+
+  def create
+    @user = User.find(params[:user_id])
+    @post = Post.new(post_params)
+    @post.user = @user
+    if @post.save
+      redirect_to user_post_url({ user_id: @user.id, id: @post.id })
+      flash[:success] = 'Post created successfully'
+    else
+      render :new, flash: { error: 'Please make sure your post is valid' }
+    end
+  end
```

Comment:
It will be nice to refactor your `create` action to have a `new` action method of it's own. Like so:
```
 def new
      @post = Post.new
    end
```
It's part of the requirement. Following rails best design. You can then use the `@post` inside the create action.